src/main/java/com/nylas/Deltas.java

```diff
@@ -0,0 +1,232 @@
+		}
+
+		ResponseBody responseBody = client.download(accessToken, url);
+		StringBuilder jsonBuilder = new StringBuilder();
+		try {
+			while(!responseBody.source().exhausted()) {
+				Buffer buffer = new Buffer();
+				long count = responseBody.source().read(buffer, 8192);
+				if(count > 1) {
+					String[] jsonStream = buffer.readUtf8().split("\n");
+					for(String json : jsonStream) {
+						jsonBuilder.append(json);
+						try {
+							Delta<? extends AccountOwnedModel> delta = adapter.fromJson(jsonBuilder.toString());
+							deltas.add(delta);
+							if(listener != null) {
+								listener.onDelta(delta);
+							}
+							jsonBuilder.setLength(0);
+						} catch (JsonEncodingException | EOFException e) {
+							// The JSON was partial, move on to the next string/chunk
+						}
+					}
+				}
+			}
+		} catch (IllegalStateException e) {
+			// Stream was closed, return the deltas we have
+		}
+
+		return deltas;
+	}
+
+	/**
+	 * Poll for a DeltaCursor since the provided cursor until a delta is received or timeout is reached
+	 * <br> {@code listener} is set to null.
+	 * <br> {@code options} is set to null.
+	 * @see Deltas#longpoll(String, int, DeltaLongPollListener, DeltaQueryOptions)
+	 */
+	public DeltaCursor longpoll(String cursor, int timeout) throws RequestFailedException, IOException {
+		return longpoll(cursor, timeout, null, null);
+	}
+
+	/**
+	 * Poll for a DeltaCursor since the provided cursor until a delta is received or timeout is reached
+	 * <br> {@code options} is set to null.
+	 * @see Deltas#longpoll(String, int, DeltaLongPollListener, DeltaQueryOptions)
+	 */
+	public DeltaCursor longpoll(String cursor, int timeout, DeltaLongPollListener listener)
+			throws RequestFailedException, IOException {
+		return longpoll(cursor, timeout, listener, null);
+	}
+
+	/**
+	 * Poll for a DeltaCursor since the provided cursor until a delta is received or timeout is reached
+	 * <br> {@code listener} is set to null.
+	 * @see Deltas#longpoll(String, int, DeltaLongPollListener, DeltaQueryOptions)
+	 */
+	public DeltaCursor longpoll(String cursor, int timeout, DeltaQueryOptions options)
+			throws RequestFailedException, IOException {
+		return longpoll(cursor, timeout, null, options);
+	}
+
+	/**
+	 * Poll for a DeltaCursor since the provided cursor until a delta is received or timeout is reached
+	 * @param cursor The cursor to start polling from
+	 * @param timeout The number of seconds to poll for before timing out
+	 * @param listener An object that implements {@link DeltaLongPollListener} to listen for a DeltaCursor
+	 * @param options Additional options for this Delta query
+	 * @return The {@link DeltaCursor} object captured while polling
+	 * @see <a href="https://developer.nylas.com/docs/api/#get/delta/longpoll">Return Long-Polling Deltas</a>
+	 */
+	public DeltaCursor longpoll(String cursor, int timeout, DeltaLongPollListener listener, DeltaQueryOptions options)
+			throws RequestFailedException, IOException {
```

Comment:
maybe validate the cursor and is there an upper bounds for the timeout value?

Reply:
Thanks!

1. Good catch, adding a checker here as well.
2. According to the docs there are not... however, in trial/error I found that you cannot have a value that exceeds 1800 seconds. Adding this constraint now, and submitted this change request to the docs team as well, good call out!